test/types/sync/ferguson/sync-record-no-default-init.chpl

```diff
@@ -0,0 +1,136 @@
+class C { }
+record R {
+  var x: int = 0;
+  var ptr: shared C = new shared C();
+  // intentionally no init() -- no default init
+  proc init(arg:int) {
+    this.x = arg;
+    writeln("init ", arg);
+  }
+  proc init=(const other: R) {
+    this.x = other.x;
+    writeln("init= ", other.x);
+  }
+  proc deinit() {
+    writeln("deinit ", x);
+  }
+}
+operator R.=(ref lhs:R, rhs:R) {
+  writeln("= ", lhs.x, " ", rhs.x);
+  lhs.x = rhs.x;
+}
+
+proc testDefaultInitSync() {
+  writeln("testDefaultInitSync");
+  var x: sync R;
+}
+testDefaultInitSync();
+
+proc testDefaultInitSyncWriteEF() {
+  writeln("testDefaultInitSyncWriteEF");
+  var x: sync R;
+  x.writeEF(new R(1));
+}
+testDefaultInitSyncWriteEF();
+
+proc testDefaultInitSyncWriteEFReadFE() {
+  writeln("testDefaultInitSyncWriteEFReadFE");
+  var x: sync R;
+  x.writeEF(new R(1));
+  var y = x.readFE();
+  writeln(y);
+}
+testDefaultInitSyncWriteEFReadFE();
+
+proc testDefaultInitSyncWriteEFReadFF() {
+  writeln("testDefaultInitSyncWriteEFReadFF");
+  var x: sync R;
+  x.writeEF(new R(1));
+  var y = x.readFF();
+  writeln(y);
+}
+testDefaultInitSyncWriteEFReadFF();
+
+proc testDefaultInitSyncWriteEFWriteFF() {
+  writeln("testDefaultInitSyncWriteEFWriteFF");
+  var x: sync R;
+  x.writeEF(new R(1));
+  x.writeFF(new R(2));
+}
+testDefaultInitSyncWriteEFWriteFF();
+
+// TODO: could get these cases working
```

Comment:
Is this TODO something that got missed before opening the PR?  Or is it a later TODO?